### PR TITLE
Convert to Monorepo

### PR DESCRIPTION
**Pull request by Turbo87**
😱 😱 😱 😱 

no, wait, hear me out!

This PR is inspired by the repository layout that https://github.com/embroider-build/embroider/ is using. Instead of using the barely maintained https://github.com/tomdale/ember-cli-addon-tests project, we instead use a monorepo layout with several fixture packages in a `test-packages` folder. 

This enables us to use these test fixtures for debugging in a much easier way than before because they can also be used outside of running the test suite. In other words you can `cd` into the folder, run `COVERAGE=true ember test` there and manually verify the results in the `coverage` folder.

It also removes the need for running `npm install` during the test suite because the initial `yarn install` would have already installed all the necessary dependencies. Locally this seems to significantly speed up the test runs, for some reason they still seem to take roughly the same time on CI though.

### CODE REVIEW

**Comment by rwjblue** [comment]
> This PR is inspired by the repository layout that embroider-build/embroider is using. Instead of using the barely maintained tomdale/ember-cli-addon-tests project, we instead use a monorepo layout with several fixture packages in a test-packages folder.

FWIW, `ember-cli-fastboot` did the same general move recently (as did `ember-engines`).

File: package.json [hunk 1]
```diff
@@ -1,123 +1,17 @@
...
-    },
-    "github": {
-      "release": true,
-      "tokenRef": "GITHUB_AUTH"
-    },
-    "npm": {
-      "publish": false
-    }
+    "rimraf": "^2.6.2"
   }
```
Comment (rwjblue) [inline package.json:16]: Should bring this back, with nearly the same configuration as https://github.com/ember-fastboot/ember-cli-fastboot/pull/783

File: packages/ember-cli-code-coverage/package.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      "tagName": "v${version}"
+    },
+    "github": {
+      "release": true,
+      "tokenRef": "GITHUB_AUTH"
+    },
+    "npm": {
+      "publish": false
+    }
+  }
```
Comment (rwjblue) [inline packages/ember-cli-code-coverage/package.json:120]: Should be removed (in favor of the changes in https://github.com/ember-fastboot/ember-cli-fastboot/pull/783).

File: test-packages/in-repo-addon-test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+'use strict';
```
Comment (rwjblue) [inline test-packages/in-repo-addon-test.js:1]: I find it somewhat odd to have this file "loose" in the same directory as other packages.

File: test-packages/in-repo-addon-test.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+const BASE_PATH = path.join(__dirname, 'my-app-with-in-repo-addon');
+
+describe('in-repo addon coverage generation', function () {
+  this.timeout(10000000);
+
+  beforeEach(async function () {
+    await rimraf(`${BASE_PATH}/coverage*`);
+    await execa('git', ['clean', '-f', 'my-app-with-in-repo-addon'], { cwd: __dirname });
+    await execa('git', ['restore', 'my-app-with-in-repo-addon'], { cwd: __dirname });
```
Comment (rwjblue) [inline test-packages/in-repo-addon-test.js:24]: Seems vaguely odd to do `git` operations in tests, can you explain what this is cleaning up for?

File: test-packages/my-app-test.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    await rimraf(`${BASE_PATH}/coverage*`);
+    await execa('git', ['clean', '-f', 'my-app'], { cwd: __dirname });
+    await execa('git', ['restore', 'my-app'], { cwd: __dirname });
+  });
+
+  afterEach(async function () {
+    await rimraf(`${BASE_PATH}/coverage*`);
+    await execa('git', ['clean', '-f', 'my-app'], { cwd: __dirname });
+    await execa('git', ['restore', 'my-app'], { cwd: __dirname });
+  });
```
Comment (rwjblue) [inline test-packages/my-app-test.js:31]: This prelude seems to be repeated a bunch of times, can we make it a util and replace this in each file with:

```js
setupTest(BASE_PATH);
```

**Review by rwjblue — CHANGES_REQUESTED** [review]
Some files should be moved back to the project root:

* `CHANGELOG.md`
* `CONTRIBUTING.md`
* `RELEASE.md`

Should add a top level version of these files:

* `README.md` (either duplicating or referencing the main `packages/ember-cli-code-coverage/README.md`)

File: packages/ember-cli-code-coverage/package.json [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  ],
+  "directories": {
+    "doc": "doc",
+    "test": "tests"
+  },
+  "scripts": {
+    "browser-test": "COVERAGE=true ember test",
+    "lint:hbs": "ember-template-lint .",
+    "lint:js": "eslint .",
+    "test": "npm run-script lint:js && npm run-script node-test"
```
Comment (kategengler) [inline packages/ember-cli-code-coverage/package.json:28]: This tries to run `node-test`, but that's gone

**Comment by kategengler** [comment]
👍 Very much in favor of this layout, thanks for doing this!

**Comment by rwjblue** [comment]
Updated to address the various review comments above.

**Comment by Turbo87** [comment]
sorry, had it on the todo list, just didn't have time yet 😩

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
> sorry, had it on the todo list, just didn't have time yet 😩

No worries! I was just helpin' out 😸

**Pull request merged**